// @pulumi/aws/identitystore/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.GroupMembership = exports.Group = exports.getUserOutput = exports.getUser = exports.getGroupOutput = exports.getGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGroup = null;
exports.getGroupOutput = null;
exports.getUser = null;
exports.getUserOutput = null;
exports.Group = null;
exports.GroupMembership = null;
exports.User = null;
utilities.lazyLoad(exports, ["getGroup", "getGroupOutput"], () => require("./getGroup"));
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
utilities.lazyLoad(exports, ["GroupMembership"], () => require("./groupMembership"));
utilities.lazyLoad(exports, ["User"], () => require("./user"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:identitystore/group:Group":
                return new exports.Group(name, undefined, { urn });
            case "aws:identitystore/groupMembership:GroupMembership":
                return new exports.GroupMembership(name, undefined, { urn });
            case "aws:identitystore/user:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "identitystore/group", _module);
pulumi.runtime.registerResourceModule("aws", "identitystore/groupMembership", _module);
pulumi.runtime.registerResourceModule("aws", "identitystore/user", _module);
