// @pulumi/aws/memorydb/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.SubnetGroup = exports.Snapshot = exports.ParameterGroup = exports.getUserOutput = exports.getUser = exports.getSubnetGroupOutput = exports.getSubnetGroup = exports.getSnapshotOutput = exports.getSnapshot = exports.getParameterGroupOutput = exports.getParameterGroup = exports.getClusterOutput = exports.getCluster = exports.getAclOutput = exports.getAcl = exports.Cluster = exports.Acl = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Acl = null;
exports.Cluster = null;
exports.getAcl = null;
exports.getAclOutput = null;
exports.getCluster = null;
exports.getClusterOutput = null;
exports.getParameterGroup = null;
exports.getParameterGroupOutput = null;
exports.getSnapshot = null;
exports.getSnapshotOutput = null;
exports.getSubnetGroup = null;
exports.getSubnetGroupOutput = null;
exports.getUser = null;
exports.getUserOutput = null;
exports.ParameterGroup = null;
exports.Snapshot = null;
exports.SubnetGroup = null;
exports.User = null;
utilities.lazyLoad(exports, ["Acl"], () => require("./acl"));
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
utilities.lazyLoad(exports, ["getAcl", "getAclOutput"], () => require("./getAcl"));
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
utilities.lazyLoad(exports, ["getParameterGroup", "getParameterGroupOutput"], () => require("./getParameterGroup"));
utilities.lazyLoad(exports, ["getSnapshot", "getSnapshotOutput"], () => require("./getSnapshot"));
utilities.lazyLoad(exports, ["getSubnetGroup", "getSubnetGroupOutput"], () => require("./getSubnetGroup"));
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
utilities.lazyLoad(exports, ["ParameterGroup"], () => require("./parameterGroup"));
utilities.lazyLoad(exports, ["Snapshot"], () => require("./snapshot"));
utilities.lazyLoad(exports, ["SubnetGroup"], () => require("./subnetGroup"));
utilities.lazyLoad(exports, ["User"], () => require("./user"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:memorydb/acl:Acl":
                return new exports.Acl(name, undefined, { urn });
            case "aws:memorydb/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "aws:memorydb/parameterGroup:ParameterGroup":
                return new exports.ParameterGroup(name, undefined, { urn });
            case "aws:memorydb/snapshot:Snapshot":
                return new exports.Snapshot(name, undefined, { urn });
            case "aws:memorydb/subnetGroup:SubnetGroup":
                return new exports.SubnetGroup(name, undefined, { urn });
            case "aws:memorydb/user:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "memorydb/acl", _module);
pulumi.runtime.registerResourceModule("aws", "memorydb/cluster", _module);
pulumi.runtime.registerResourceModule("aws", "memorydb/parameterGroup", _module);
pulumi.runtime.registerResourceModule("aws", "memorydb/snapshot", _module);
pulumi.runtime.registerResourceModule("aws", "memorydb/subnetGroup", _module);
pulumi.runtime.registerResourceModule("aws", "memorydb/user", _module);
