// @pulumi/aws/cognito/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserPoolUICustomization = exports.UserPoolDomain = exports.UserPoolClient = exports.UserPool = exports.UserInGroup = exports.UserGroup = exports.User = exports.RiskConfiguration = exports.ResourceServer = exports.IdentityProvider = exports.IdentityPoolRoleAttachment = exports.IdentityPoolProviderPrincipalTag = exports.IdentityPool = exports.getUserPoolsOutput = exports.getUserPools = exports.getUserPoolSigningCertificateOutput = exports.getUserPoolSigningCertificate = exports.getUserPoolClientsOutput = exports.getUserPoolClients = exports.getUserPoolClientOutput = exports.getUserPoolClient = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getUserPoolClient = null;
exports.getUserPoolClientOutput = null;
exports.getUserPoolClients = null;
exports.getUserPoolClientsOutput = null;
exports.getUserPoolSigningCertificate = null;
exports.getUserPoolSigningCertificateOutput = null;
exports.getUserPools = null;
exports.getUserPoolsOutput = null;
exports.IdentityPool = null;
exports.IdentityPoolProviderPrincipalTag = null;
exports.IdentityPoolRoleAttachment = null;
exports.IdentityProvider = null;
exports.ResourceServer = null;
exports.RiskConfiguration = null;
exports.User = null;
exports.UserGroup = null;
exports.UserInGroup = null;
exports.UserPool = null;
exports.UserPoolClient = null;
exports.UserPoolDomain = null;
exports.UserPoolUICustomization = null;
utilities.lazyLoad(exports, ["getUserPoolClient", "getUserPoolClientOutput"], () => require("./getUserPoolClient"));
utilities.lazyLoad(exports, ["getUserPoolClients", "getUserPoolClientsOutput"], () => require("./getUserPoolClients"));
utilities.lazyLoad(exports, ["getUserPoolSigningCertificate", "getUserPoolSigningCertificateOutput"], () => require("./getUserPoolSigningCertificate"));
utilities.lazyLoad(exports, ["getUserPools", "getUserPoolsOutput"], () => require("./getUserPools"));
utilities.lazyLoad(exports, ["IdentityPool"], () => require("./identityPool"));
utilities.lazyLoad(exports, ["IdentityPoolProviderPrincipalTag"], () => require("./identityPoolProviderPrincipalTag"));
utilities.lazyLoad(exports, ["IdentityPoolRoleAttachment"], () => require("./identityPoolRoleAttachment"));
utilities.lazyLoad(exports, ["IdentityProvider"], () => require("./identityProvider"));
utilities.lazyLoad(exports, ["ResourceServer"], () => require("./resourceServer"));
utilities.lazyLoad(exports, ["RiskConfiguration"], () => require("./riskConfiguration"));
utilities.lazyLoad(exports, ["User"], () => require("./user"));
utilities.lazyLoad(exports, ["UserGroup"], () => require("./userGroup"));
utilities.lazyLoad(exports, ["UserInGroup"], () => require("./userInGroup"));
utilities.lazyLoad(exports, ["UserPool"], () => require("./userPool"));
utilities.lazyLoad(exports, ["UserPoolClient"], () => require("./userPoolClient"));
utilities.lazyLoad(exports, ["UserPoolDomain"], () => require("./userPoolDomain"));
utilities.lazyLoad(exports, ["UserPoolUICustomization"], () => require("./userPoolUICustomization"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:cognito/identityPool:IdentityPool":
                return new exports.IdentityPool(name, undefined, { urn });
            case "aws:cognito/identityPoolProviderPrincipalTag:IdentityPoolProviderPrincipalTag":
                return new exports.IdentityPoolProviderPrincipalTag(name, undefined, { urn });
            case "aws:cognito/identityPoolRoleAttachment:IdentityPoolRoleAttachment":
                return new exports.IdentityPoolRoleAttachment(name, undefined, { urn });
            case "aws:cognito/identityProvider:IdentityProvider":
                return new exports.IdentityProvider(name, undefined, { urn });
            case "aws:cognito/resourceServer:ResourceServer":
                return new exports.ResourceServer(name, undefined, { urn });
            case "aws:cognito/riskConfiguration:RiskConfiguration":
                return new exports.RiskConfiguration(name, undefined, { urn });
            case "aws:cognito/user:User":
                return new exports.User(name, undefined, { urn });
            case "aws:cognito/userGroup:UserGroup":
                return new exports.UserGroup(name, undefined, { urn });
            case "aws:cognito/userInGroup:UserInGroup":
                return new exports.UserInGroup(name, undefined, { urn });
            case "aws:cognito/userPool:UserPool":
                return new exports.UserPool(name, undefined, { urn });
            case "aws:cognito/userPoolClient:UserPoolClient":
                return new exports.UserPoolClient(name, undefined, { urn });
            case "aws:cognito/userPoolDomain:UserPoolDomain":
                return new exports.UserPoolDomain(name, undefined, { urn });
            case "aws:cognito/userPoolUICustomization:UserPoolUICustomization":
                return new exports.UserPoolUICustomization(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "cognito/identityPool", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/identityPoolProviderPrincipalTag", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/identityPoolRoleAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/identityProvider", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/resourceServer", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/riskConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/user", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/userGroup", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/userInGroup", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/userPool", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/userPoolClient", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/userPoolDomain", _module);
pulumi.runtime.registerResourceModule("aws", "cognito/userPoolUICustomization", _module);
