// @pulumi/aws/chime/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoiceConnectorTerminationCredentials = exports.VoiceConnectorTermination = exports.VoiceConnectorStreaming = exports.VoiceConnectorOrganization = exports.VoiceConnectorLogging = exports.VoiceConnectorGroup = exports.VoiceConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.VoiceConnector = null;
exports.VoiceConnectorGroup = null;
exports.VoiceConnectorLogging = null;
exports.VoiceConnectorOrganization = null;
exports.VoiceConnectorStreaming = null;
exports.VoiceConnectorTermination = null;
exports.VoiceConnectorTerminationCredentials = null;
utilities.lazyLoad(exports, ["VoiceConnector"], () => require("./voiceConnector"));
utilities.lazyLoad(exports, ["VoiceConnectorGroup"], () => require("./voiceConnectorGroup"));
utilities.lazyLoad(exports, ["VoiceConnectorLogging"], () => require("./voiceConnectorLogging"));
utilities.lazyLoad(exports, ["VoiceConnectorOrganization"], () => require("./voiceConnectorOrganization"));
utilities.lazyLoad(exports, ["VoiceConnectorStreaming"], () => require("./voiceConnectorStreaming"));
utilities.lazyLoad(exports, ["VoiceConnectorTermination"], () => require("./voiceConnectorTermination"));
utilities.lazyLoad(exports, ["VoiceConnectorTerminationCredentials"], () => require("./voiceConnectorTerminationCredentials"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:chime/voiceConnector:VoiceConnector":
                return new exports.VoiceConnector(name, undefined, { urn });
            case "aws:chime/voiceConnectorGroup:VoiceConnectorGroup":
                return new exports.VoiceConnectorGroup(name, undefined, { urn });
            case "aws:chime/voiceConnectorLogging:VoiceConnectorLogging":
                return new exports.VoiceConnectorLogging(name, undefined, { urn });
            case "aws:chime/voiceConnectorOrganization:VoiceConnectorOrganization":
                return new exports.VoiceConnectorOrganization(name, undefined, { urn });
            case "aws:chime/voiceConnectorStreaming:VoiceConnectorStreaming":
                return new exports.VoiceConnectorStreaming(name, undefined, { urn });
            case "aws:chime/voiceConnectorTermination:VoiceConnectorTermination":
                return new exports.VoiceConnectorTermination(name, undefined, { urn });
            case "aws:chime/voiceConnectorTerminationCredentials:VoiceConnectorTerminationCredentials":
                return new exports.VoiceConnectorTerminationCredentials(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnector", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorGroup", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorLogging", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorOrganization", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorStreaming", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorTermination", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorTerminationCredentials", _module);
