// @pulumi/aws/opsworks/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserProfile = exports.StaticWebLayer = exports.Stack = exports.RdsDbInstance = exports.RailsAppLayer = exports.PhpAppLayer = exports.Permission = exports.NodejsAppLayer = exports.MysqlLayer = exports.MemcachedLayer = exports.JavaAppLayer = exports.Instance = exports.HaproxyLayer = exports.GangliaLayer = exports.EcsClusterLayer = exports.CustomLayer = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Application = null;
exports.CustomLayer = null;
exports.EcsClusterLayer = null;
exports.GangliaLayer = null;
exports.HaproxyLayer = null;
exports.Instance = null;
exports.JavaAppLayer = null;
exports.MemcachedLayer = null;
exports.MysqlLayer = null;
exports.NodejsAppLayer = null;
exports.Permission = null;
exports.PhpAppLayer = null;
exports.RailsAppLayer = null;
exports.RdsDbInstance = null;
exports.Stack = null;
exports.StaticWebLayer = null;
exports.UserProfile = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
utilities.lazyLoad(exports, ["CustomLayer"], () => require("./customLayer"));
utilities.lazyLoad(exports, ["EcsClusterLayer"], () => require("./ecsClusterLayer"));
utilities.lazyLoad(exports, ["GangliaLayer"], () => require("./gangliaLayer"));
utilities.lazyLoad(exports, ["HaproxyLayer"], () => require("./haproxyLayer"));
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
utilities.lazyLoad(exports, ["JavaAppLayer"], () => require("./javaAppLayer"));
utilities.lazyLoad(exports, ["MemcachedLayer"], () => require("./memcachedLayer"));
utilities.lazyLoad(exports, ["MysqlLayer"], () => require("./mysqlLayer"));
utilities.lazyLoad(exports, ["NodejsAppLayer"], () => require("./nodejsAppLayer"));
utilities.lazyLoad(exports, ["Permission"], () => require("./permission"));
utilities.lazyLoad(exports, ["PhpAppLayer"], () => require("./phpAppLayer"));
utilities.lazyLoad(exports, ["RailsAppLayer"], () => require("./railsAppLayer"));
utilities.lazyLoad(exports, ["RdsDbInstance"], () => require("./rdsDbInstance"));
utilities.lazyLoad(exports, ["Stack"], () => require("./stack"));
utilities.lazyLoad(exports, ["StaticWebLayer"], () => require("./staticWebLayer"));
utilities.lazyLoad(exports, ["UserProfile"], () => require("./userProfile"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:opsworks/application:Application":
                return new exports.Application(name, undefined, { urn });
            case "aws:opsworks/customLayer:CustomLayer":
                return new exports.CustomLayer(name, undefined, { urn });
            case "aws:opsworks/ecsClusterLayer:EcsClusterLayer":
                return new exports.EcsClusterLayer(name, undefined, { urn });
            case "aws:opsworks/gangliaLayer:GangliaLayer":
                return new exports.GangliaLayer(name, undefined, { urn });
            case "aws:opsworks/haproxyLayer:HaproxyLayer":
                return new exports.HaproxyLayer(name, undefined, { urn });
            case "aws:opsworks/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "aws:opsworks/javaAppLayer:JavaAppLayer":
                return new exports.JavaAppLayer(name, undefined, { urn });
            case "aws:opsworks/memcachedLayer:MemcachedLayer":
                return new exports.MemcachedLayer(name, undefined, { urn });
            case "aws:opsworks/mysqlLayer:MysqlLayer":
                return new exports.MysqlLayer(name, undefined, { urn });
            case "aws:opsworks/nodejsAppLayer:NodejsAppLayer":
                return new exports.NodejsAppLayer(name, undefined, { urn });
            case "aws:opsworks/permission:Permission":
                return new exports.Permission(name, undefined, { urn });
            case "aws:opsworks/phpAppLayer:PhpAppLayer":
                return new exports.PhpAppLayer(name, undefined, { urn });
            case "aws:opsworks/railsAppLayer:RailsAppLayer":
                return new exports.RailsAppLayer(name, undefined, { urn });
            case "aws:opsworks/rdsDbInstance:RdsDbInstance":
                return new exports.RdsDbInstance(name, undefined, { urn });
            case "aws:opsworks/stack:Stack":
                return new exports.Stack(name, undefined, { urn });
            case "aws:opsworks/staticWebLayer:StaticWebLayer":
                return new exports.StaticWebLayer(name, undefined, { urn });
            case "aws:opsworks/userProfile:UserProfile":
                return new exports.UserProfile(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "opsworks/application", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/customLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/ecsClusterLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/gangliaLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/haproxyLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/instance", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/javaAppLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/memcachedLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/mysqlLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/nodejsAppLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/permission", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/phpAppLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/railsAppLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/rdsDbInstance", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/stack", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/staticWebLayer", _module);
pulumi.runtime.registerResourceModule("aws", "opsworks/userProfile", _module);
