// @pulumi/aws/quicksight/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.GroupMembership = exports.Group = exports.DataSource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DataSource = null;
exports.Group = null;
exports.GroupMembership = null;
exports.User = null;
utilities.lazyLoad(exports, ["DataSource"], () => require("./dataSource"));
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
utilities.lazyLoad(exports, ["GroupMembership"], () => require("./groupMembership"));
utilities.lazyLoad(exports, ["User"], () => require("./user"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:quicksight/dataSource:DataSource":
                return new exports.DataSource(name, undefined, { urn });
            case "aws:quicksight/group:Group":
                return new exports.Group(name, undefined, { urn });
            case "aws:quicksight/groupMembership:GroupMembership":
                return new exports.GroupMembership(name, undefined, { urn });
            case "aws:quicksight/user:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "quicksight/dataSource", _module);
pulumi.runtime.registerResourceModule("aws", "quicksight/group", _module);
pulumi.runtime.registerResourceModule("aws", "quicksight/groupMembership", _module);
pulumi.runtime.registerResourceModule("aws", "quicksight/user", _module);
