// @pulumi/aws/secretsmanager/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecretVersion = exports.SecretRotation = exports.SecretPolicy = exports.Secret = exports.getSecretsOutput = exports.getSecrets = exports.getSecretVersionOutput = exports.getSecretVersion = exports.getSecretRotationOutput = exports.getSecretRotation = exports.getSecretOutput = exports.getSecret = exports.getRandomPasswordOutput = exports.getRandomPassword = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getRandomPassword = null;
exports.getRandomPasswordOutput = null;
exports.getSecret = null;
exports.getSecretOutput = null;
exports.getSecretRotation = null;
exports.getSecretRotationOutput = null;
exports.getSecretVersion = null;
exports.getSecretVersionOutput = null;
exports.getSecrets = null;
exports.getSecretsOutput = null;
exports.Secret = null;
exports.SecretPolicy = null;
exports.SecretRotation = null;
exports.SecretVersion = null;
utilities.lazyLoad(exports, ["getRandomPassword", "getRandomPasswordOutput"], () => require("./getRandomPassword"));
utilities.lazyLoad(exports, ["getSecret", "getSecretOutput"], () => require("./getSecret"));
utilities.lazyLoad(exports, ["getSecretRotation", "getSecretRotationOutput"], () => require("./getSecretRotation"));
utilities.lazyLoad(exports, ["getSecretVersion", "getSecretVersionOutput"], () => require("./getSecretVersion"));
utilities.lazyLoad(exports, ["getSecrets", "getSecretsOutput"], () => require("./getSecrets"));
utilities.lazyLoad(exports, ["Secret"], () => require("./secret"));
utilities.lazyLoad(exports, ["SecretPolicy"], () => require("./secretPolicy"));
utilities.lazyLoad(exports, ["SecretRotation"], () => require("./secretRotation"));
utilities.lazyLoad(exports, ["SecretVersion"], () => require("./secretVersion"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:secretsmanager/secret:Secret":
                return new exports.Secret(name, undefined, { urn });
            case "aws:secretsmanager/secretPolicy:SecretPolicy":
                return new exports.SecretPolicy(name, undefined, { urn });
            case "aws:secretsmanager/secretRotation:SecretRotation":
                return new exports.SecretRotation(name, undefined, { urn });
            case "aws:secretsmanager/secretVersion:SecretVersion":
                return new exports.SecretVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "secretsmanager/secret", _module);
pulumi.runtime.registerResourceModule("aws", "secretsmanager/secretPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "secretsmanager/secretRotation", _module);
pulumi.runtime.registerResourceModule("aws", "secretsmanager/secretVersion", _module);
