// @pulumi/aws/sns/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicSubscription = exports.TopicPolicy = exports.Topic = exports.SmsPreferences = exports.PlatformApplication = exports.getTopicOutput = exports.getTopic = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getTopic = null;
exports.getTopicOutput = null;
exports.PlatformApplication = null;
exports.SmsPreferences = null;
__exportStar(require("./snsMixins"), exports);
exports.Topic = null;
exports.TopicPolicy = null;
exports.TopicSubscription = null;
utilities.lazyLoad(exports, ["getTopic", "getTopicOutput"], () => require("./getTopic"));
utilities.lazyLoad(exports, ["PlatformApplication"], () => require("./platformApplication"));
utilities.lazyLoad(exports, ["SmsPreferences"], () => require("./smsPreferences"));
utilities.lazyLoad(exports, ["Topic"], () => require("./topic"));
utilities.lazyLoad(exports, ["TopicPolicy"], () => require("./topicPolicy"));
utilities.lazyLoad(exports, ["TopicSubscription"], () => require("./topicSubscription"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:sns/platformApplication:PlatformApplication":
                return new exports.PlatformApplication(name, undefined, { urn });
            case "aws:sns/smsPreferences:SmsPreferences":
                return new exports.SmsPreferences(name, undefined, { urn });
            case "aws:sns/topic:Topic":
                return new exports.Topic(name, undefined, { urn });
            case "aws:sns/topicPolicy:TopicPolicy":
                return new exports.TopicPolicy(name, undefined, { urn });
            case "aws:sns/topicSubscription:TopicSubscription":
                return new exports.TopicSubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "sns/platformApplication", _module);
pulumi.runtime.registerResourceModule("aws", "sns/smsPreferences", _module);
pulumi.runtime.registerResourceModule("aws", "sns/topic", _module);
pulumi.runtime.registerResourceModule("aws", "sns/topicPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "sns/topicSubscription", _module);
