// @pulumi/aws/outposts/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSites = exports.getSiteOutput = exports.getSite = exports.getOutpostsOutput = exports.getOutposts = exports.getOutpostInstanceTypesOutput = exports.getOutpostInstanceTypes = exports.getOutpostInstanceTypeOutput = exports.getOutpostInstanceType = exports.getOutpostOutput = exports.getOutpost = exports.getAssetsOutput = exports.getAssets = exports.getAssetOutput = exports.getAsset = void 0;
const utilities = require("../utilities");
exports.getAsset = null;
exports.getAssetOutput = null;
exports.getAssets = null;
exports.getAssetsOutput = null;
exports.getOutpost = null;
exports.getOutpostOutput = null;
exports.getOutpostInstanceType = null;
exports.getOutpostInstanceTypeOutput = null;
exports.getOutpostInstanceTypes = null;
exports.getOutpostInstanceTypesOutput = null;
exports.getOutposts = null;
exports.getOutpostsOutput = null;
exports.getSite = null;
exports.getSiteOutput = null;
exports.getSites = null;
utilities.lazyLoad(exports, ["getAsset", "getAssetOutput"], () => require("./getAsset"));
utilities.lazyLoad(exports, ["getAssets", "getAssetsOutput"], () => require("./getAssets"));
utilities.lazyLoad(exports, ["getOutpost", "getOutpostOutput"], () => require("./getOutpost"));
utilities.lazyLoad(exports, ["getOutpostInstanceType", "getOutpostInstanceTypeOutput"], () => require("./getOutpostInstanceType"));
utilities.lazyLoad(exports, ["getOutpostInstanceTypes", "getOutpostInstanceTypesOutput"], () => require("./getOutpostInstanceTypes"));
utilities.lazyLoad(exports, ["getOutposts", "getOutpostsOutput"], () => require("./getOutposts"));
utilities.lazyLoad(exports, ["getSite", "getSiteOutput"], () => require("./getSite"));
utilities.lazyLoad(exports, ["getSites"], () => require("./getSites"));
