// @pulumi/aws/directconnect/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransitVirtualInterface = exports.PublicVirtualInterface = exports.PrivateVirtualInterface = exports.LinkAggregationGroup = exports.HostedTransitVirtualInterfaceAcceptor = exports.HostedTransitVirtualInterface = exports.HostedPublicVirtualInterfaceAccepter = exports.HostedPublicVirtualInterface = exports.HostedPrivateVirtualInterfaceAccepter = exports.HostedPrivateVirtualInterface = exports.HostedConnection = exports.getLocations = exports.getLocationOutput = exports.getLocation = exports.getGatewayOutput = exports.getGateway = exports.getConnectionOutput = exports.getConnection = exports.GatewayAssociationProposal = exports.GatewayAssociation = exports.Gateway = exports.ConnectionConfirmation = exports.ConnectionAssociation = exports.Connection = exports.BgpPeer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.BgpPeer = null;
exports.Connection = null;
exports.ConnectionAssociation = null;
exports.ConnectionConfirmation = null;
exports.Gateway = null;
exports.GatewayAssociation = null;
exports.GatewayAssociationProposal = null;
exports.getConnection = null;
exports.getConnectionOutput = null;
exports.getGateway = null;
exports.getGatewayOutput = null;
exports.getLocation = null;
exports.getLocationOutput = null;
exports.getLocations = null;
exports.HostedConnection = null;
exports.HostedPrivateVirtualInterface = null;
exports.HostedPrivateVirtualInterfaceAccepter = null;
exports.HostedPublicVirtualInterface = null;
exports.HostedPublicVirtualInterfaceAccepter = null;
exports.HostedTransitVirtualInterface = null;
exports.HostedTransitVirtualInterfaceAcceptor = null;
exports.LinkAggregationGroup = null;
exports.PrivateVirtualInterface = null;
exports.PublicVirtualInterface = null;
exports.TransitVirtualInterface = null;
utilities.lazyLoad(exports, ["BgpPeer"], () => require("./bgpPeer"));
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
utilities.lazyLoad(exports, ["ConnectionAssociation"], () => require("./connectionAssociation"));
utilities.lazyLoad(exports, ["ConnectionConfirmation"], () => require("./connectionConfirmation"));
utilities.lazyLoad(exports, ["Gateway"], () => require("./gateway"));
utilities.lazyLoad(exports, ["GatewayAssociation"], () => require("./gatewayAssociation"));
utilities.lazyLoad(exports, ["GatewayAssociationProposal"], () => require("./gatewayAssociationProposal"));
utilities.lazyLoad(exports, ["getConnection", "getConnectionOutput"], () => require("./getConnection"));
utilities.lazyLoad(exports, ["getGateway", "getGatewayOutput"], () => require("./getGateway"));
utilities.lazyLoad(exports, ["getLocation", "getLocationOutput"], () => require("./getLocation"));
utilities.lazyLoad(exports, ["getLocations"], () => require("./getLocations"));
utilities.lazyLoad(exports, ["HostedConnection"], () => require("./hostedConnection"));
utilities.lazyLoad(exports, ["HostedPrivateVirtualInterface"], () => require("./hostedPrivateVirtualInterface"));
utilities.lazyLoad(exports, ["HostedPrivateVirtualInterfaceAccepter"], () => require("./hostedPrivateVirtualInterfaceAccepter"));
utilities.lazyLoad(exports, ["HostedPublicVirtualInterface"], () => require("./hostedPublicVirtualInterface"));
utilities.lazyLoad(exports, ["HostedPublicVirtualInterfaceAccepter"], () => require("./hostedPublicVirtualInterfaceAccepter"));
utilities.lazyLoad(exports, ["HostedTransitVirtualInterface"], () => require("./hostedTransitVirtualInterface"));
utilities.lazyLoad(exports, ["HostedTransitVirtualInterfaceAcceptor"], () => require("./hostedTransitVirtualInterfaceAcceptor"));
utilities.lazyLoad(exports, ["LinkAggregationGroup"], () => require("./linkAggregationGroup"));
utilities.lazyLoad(exports, ["PrivateVirtualInterface"], () => require("./privateVirtualInterface"));
utilities.lazyLoad(exports, ["PublicVirtualInterface"], () => require("./publicVirtualInterface"));
utilities.lazyLoad(exports, ["TransitVirtualInterface"], () => require("./transitVirtualInterface"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:directconnect/bgpPeer:BgpPeer":
                return new exports.BgpPeer(name, undefined, { urn });
            case "aws:directconnect/connection:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "aws:directconnect/connectionAssociation:ConnectionAssociation":
                return new exports.ConnectionAssociation(name, undefined, { urn });
            case "aws:directconnect/connectionConfirmation:ConnectionConfirmation":
                return new exports.ConnectionConfirmation(name, undefined, { urn });
            case "aws:directconnect/gateway:Gateway":
                return new exports.Gateway(name, undefined, { urn });
            case "aws:directconnect/gatewayAssociation:GatewayAssociation":
                return new exports.GatewayAssociation(name, undefined, { urn });
            case "aws:directconnect/gatewayAssociationProposal:GatewayAssociationProposal":
                return new exports.GatewayAssociationProposal(name, undefined, { urn });
            case "aws:directconnect/hostedConnection:HostedConnection":
                return new exports.HostedConnection(name, undefined, { urn });
            case "aws:directconnect/hostedPrivateVirtualInterface:HostedPrivateVirtualInterface":
                return new exports.HostedPrivateVirtualInterface(name, undefined, { urn });
            case "aws:directconnect/hostedPrivateVirtualInterfaceAccepter:HostedPrivateVirtualInterfaceAccepter":
                return new exports.HostedPrivateVirtualInterfaceAccepter(name, undefined, { urn });
            case "aws:directconnect/hostedPublicVirtualInterface:HostedPublicVirtualInterface":
                return new exports.HostedPublicVirtualInterface(name, undefined, { urn });
            case "aws:directconnect/hostedPublicVirtualInterfaceAccepter:HostedPublicVirtualInterfaceAccepter":
                return new exports.HostedPublicVirtualInterfaceAccepter(name, undefined, { urn });
            case "aws:directconnect/hostedTransitVirtualInterface:HostedTransitVirtualInterface":
                return new exports.HostedTransitVirtualInterface(name, undefined, { urn });
            case "aws:directconnect/hostedTransitVirtualInterfaceAcceptor:HostedTransitVirtualInterfaceAcceptor":
                return new exports.HostedTransitVirtualInterfaceAcceptor(name, undefined, { urn });
            case "aws:directconnect/linkAggregationGroup:LinkAggregationGroup":
                return new exports.LinkAggregationGroup(name, undefined, { urn });
            case "aws:directconnect/privateVirtualInterface:PrivateVirtualInterface":
                return new exports.PrivateVirtualInterface(name, undefined, { urn });
            case "aws:directconnect/publicVirtualInterface:PublicVirtualInterface":
                return new exports.PublicVirtualInterface(name, undefined, { urn });
            case "aws:directconnect/transitVirtualInterface:TransitVirtualInterface":
                return new exports.TransitVirtualInterface(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "directconnect/bgpPeer", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/connection", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/connectionAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/connectionConfirmation", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/gateway", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/gatewayAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/gatewayAssociationProposal", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/hostedConnection", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/hostedPrivateVirtualInterface", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/hostedPrivateVirtualInterfaceAccepter", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/hostedPublicVirtualInterface", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/hostedPublicVirtualInterfaceAccepter", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/hostedTransitVirtualInterface", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/hostedTransitVirtualInterfaceAcceptor", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/linkAggregationGroup", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/privateVirtualInterface", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/publicVirtualInterface", _module);
pulumi.runtime.registerResourceModule("aws", "directconnect/transitVirtualInterface", _module);
