// @pulumi/aws/connect/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vocabulary = exports.UserHierarchyStructure = exports.UserHierarchyGroup = exports.User = exports.SecurityProfile = exports.RoutingProfile = exports.QuickConnect = exports.Queue = exports.LambdaFunctionAssociation = exports.InstanceStorageConfig = exports.Instance = exports.HoursOfOperation = exports.getUserHierarchyStructureOutput = exports.getUserHierarchyStructure = exports.getUserHierarchyGroupOutput = exports.getUserHierarchyGroup = exports.getSecurityProfileOutput = exports.getSecurityProfile = exports.getRoutingProfileOutput = exports.getRoutingProfile = exports.getQuickConnectOutput = exports.getQuickConnect = exports.getQueueOutput = exports.getQueue = exports.getPromptOutput = exports.getPrompt = exports.getLambdaFunctionAssociationOutput = exports.getLambdaFunctionAssociation = exports.getInstanceOutput = exports.getInstance = exports.getHoursOfOperationOutput = exports.getHoursOfOperation = exports.getContactFlowModuleOutput = exports.getContactFlowModule = exports.getContactFlowOutput = exports.getContactFlow = exports.getBotAssociationOutput = exports.getBotAssociation = exports.ContactFlowModule = exports.ContactFlow = exports.BotAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.BotAssociation = null;
exports.ContactFlow = null;
exports.ContactFlowModule = null;
exports.getBotAssociation = null;
exports.getBotAssociationOutput = null;
exports.getContactFlow = null;
exports.getContactFlowOutput = null;
exports.getContactFlowModule = null;
exports.getContactFlowModuleOutput = null;
exports.getHoursOfOperation = null;
exports.getHoursOfOperationOutput = null;
exports.getInstance = null;
exports.getInstanceOutput = null;
exports.getLambdaFunctionAssociation = null;
exports.getLambdaFunctionAssociationOutput = null;
exports.getPrompt = null;
exports.getPromptOutput = null;
exports.getQueue = null;
exports.getQueueOutput = null;
exports.getQuickConnect = null;
exports.getQuickConnectOutput = null;
exports.getRoutingProfile = null;
exports.getRoutingProfileOutput = null;
exports.getSecurityProfile = null;
exports.getSecurityProfileOutput = null;
exports.getUserHierarchyGroup = null;
exports.getUserHierarchyGroupOutput = null;
exports.getUserHierarchyStructure = null;
exports.getUserHierarchyStructureOutput = null;
exports.HoursOfOperation = null;
exports.Instance = null;
exports.InstanceStorageConfig = null;
exports.LambdaFunctionAssociation = null;
exports.Queue = null;
exports.QuickConnect = null;
exports.RoutingProfile = null;
exports.SecurityProfile = null;
exports.User = null;
exports.UserHierarchyGroup = null;
exports.UserHierarchyStructure = null;
exports.Vocabulary = null;
utilities.lazyLoad(exports, ["BotAssociation"], () => require("./botAssociation"));
utilities.lazyLoad(exports, ["ContactFlow"], () => require("./contactFlow"));
utilities.lazyLoad(exports, ["ContactFlowModule"], () => require("./contactFlowModule"));
utilities.lazyLoad(exports, ["getBotAssociation", "getBotAssociationOutput"], () => require("./getBotAssociation"));
utilities.lazyLoad(exports, ["getContactFlow", "getContactFlowOutput"], () => require("./getContactFlow"));
utilities.lazyLoad(exports, ["getContactFlowModule", "getContactFlowModuleOutput"], () => require("./getContactFlowModule"));
utilities.lazyLoad(exports, ["getHoursOfOperation", "getHoursOfOperationOutput"], () => require("./getHoursOfOperation"));
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
utilities.lazyLoad(exports, ["getLambdaFunctionAssociation", "getLambdaFunctionAssociationOutput"], () => require("./getLambdaFunctionAssociation"));
utilities.lazyLoad(exports, ["getPrompt", "getPromptOutput"], () => require("./getPrompt"));
utilities.lazyLoad(exports, ["getQueue", "getQueueOutput"], () => require("./getQueue"));
utilities.lazyLoad(exports, ["getQuickConnect", "getQuickConnectOutput"], () => require("./getQuickConnect"));
utilities.lazyLoad(exports, ["getRoutingProfile", "getRoutingProfileOutput"], () => require("./getRoutingProfile"));
utilities.lazyLoad(exports, ["getSecurityProfile", "getSecurityProfileOutput"], () => require("./getSecurityProfile"));
utilities.lazyLoad(exports, ["getUserHierarchyGroup", "getUserHierarchyGroupOutput"], () => require("./getUserHierarchyGroup"));
utilities.lazyLoad(exports, ["getUserHierarchyStructure", "getUserHierarchyStructureOutput"], () => require("./getUserHierarchyStructure"));
utilities.lazyLoad(exports, ["HoursOfOperation"], () => require("./hoursOfOperation"));
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
utilities.lazyLoad(exports, ["InstanceStorageConfig"], () => require("./instanceStorageConfig"));
utilities.lazyLoad(exports, ["LambdaFunctionAssociation"], () => require("./lambdaFunctionAssociation"));
utilities.lazyLoad(exports, ["Queue"], () => require("./queue"));
utilities.lazyLoad(exports, ["QuickConnect"], () => require("./quickConnect"));
utilities.lazyLoad(exports, ["RoutingProfile"], () => require("./routingProfile"));
utilities.lazyLoad(exports, ["SecurityProfile"], () => require("./securityProfile"));
utilities.lazyLoad(exports, ["User"], () => require("./user"));
utilities.lazyLoad(exports, ["UserHierarchyGroup"], () => require("./userHierarchyGroup"));
utilities.lazyLoad(exports, ["UserHierarchyStructure"], () => require("./userHierarchyStructure"));
utilities.lazyLoad(exports, ["Vocabulary"], () => require("./vocabulary"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:connect/botAssociation:BotAssociation":
                return new exports.BotAssociation(name, undefined, { urn });
            case "aws:connect/contactFlow:ContactFlow":
                return new exports.ContactFlow(name, undefined, { urn });
            case "aws:connect/contactFlowModule:ContactFlowModule":
                return new exports.ContactFlowModule(name, undefined, { urn });
            case "aws:connect/hoursOfOperation:HoursOfOperation":
                return new exports.HoursOfOperation(name, undefined, { urn });
            case "aws:connect/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "aws:connect/instanceStorageConfig:InstanceStorageConfig":
                return new exports.InstanceStorageConfig(name, undefined, { urn });
            case "aws:connect/lambdaFunctionAssociation:LambdaFunctionAssociation":
                return new exports.LambdaFunctionAssociation(name, undefined, { urn });
            case "aws:connect/queue:Queue":
                return new exports.Queue(name, undefined, { urn });
            case "aws:connect/quickConnect:QuickConnect":
                return new exports.QuickConnect(name, undefined, { urn });
            case "aws:connect/routingProfile:RoutingProfile":
                return new exports.RoutingProfile(name, undefined, { urn });
            case "aws:connect/securityProfile:SecurityProfile":
                return new exports.SecurityProfile(name, undefined, { urn });
            case "aws:connect/user:User":
                return new exports.User(name, undefined, { urn });
            case "aws:connect/userHierarchyGroup:UserHierarchyGroup":
                return new exports.UserHierarchyGroup(name, undefined, { urn });
            case "aws:connect/userHierarchyStructure:UserHierarchyStructure":
                return new exports.UserHierarchyStructure(name, undefined, { urn });
            case "aws:connect/vocabulary:Vocabulary":
                return new exports.Vocabulary(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "connect/botAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "connect/contactFlow", _module);
pulumi.runtime.registerResourceModule("aws", "connect/contactFlowModule", _module);
pulumi.runtime.registerResourceModule("aws", "connect/hoursOfOperation", _module);
pulumi.runtime.registerResourceModule("aws", "connect/instance", _module);
pulumi.runtime.registerResourceModule("aws", "connect/instanceStorageConfig", _module);
pulumi.runtime.registerResourceModule("aws", "connect/lambdaFunctionAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "connect/queue", _module);
pulumi.runtime.registerResourceModule("aws", "connect/quickConnect", _module);
pulumi.runtime.registerResourceModule("aws", "connect/routingProfile", _module);
pulumi.runtime.registerResourceModule("aws", "connect/securityProfile", _module);
pulumi.runtime.registerResourceModule("aws", "connect/user", _module);
pulumi.runtime.registerResourceModule("aws", "connect/userHierarchyGroup", _module);
pulumi.runtime.registerResourceModule("aws", "connect/userHierarchyStructure", _module);
pulumi.runtime.registerResourceModule("aws", "connect/vocabulary", _module);
