// @pulumi/aws/eks/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeGroup = exports.IdentityProviderConfig = exports.getNodeGroupsOutput = exports.getNodeGroups = exports.getNodeGroupOutput = exports.getNodeGroup = exports.getClusters = exports.getClusterAuthOutput = exports.getClusterAuth = exports.getClusterOutput = exports.getCluster = exports.getAddonVersionOutput = exports.getAddonVersion = exports.getAddonOutput = exports.getAddon = exports.FargateProfile = exports.Cluster = exports.Addon = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Addon = null;
exports.Cluster = null;
exports.FargateProfile = null;
exports.getAddon = null;
exports.getAddonOutput = null;
exports.getAddonVersion = null;
exports.getAddonVersionOutput = null;
exports.getCluster = null;
exports.getClusterOutput = null;
exports.getClusterAuth = null;
exports.getClusterAuthOutput = null;
exports.getClusters = null;
exports.getNodeGroup = null;
exports.getNodeGroupOutput = null;
exports.getNodeGroups = null;
exports.getNodeGroupsOutput = null;
exports.IdentityProviderConfig = null;
exports.NodeGroup = null;
utilities.lazyLoad(exports, ["Addon"], () => require("./addon"));
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
utilities.lazyLoad(exports, ["FargateProfile"], () => require("./fargateProfile"));
utilities.lazyLoad(exports, ["getAddon", "getAddonOutput"], () => require("./getAddon"));
utilities.lazyLoad(exports, ["getAddonVersion", "getAddonVersionOutput"], () => require("./getAddonVersion"));
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
utilities.lazyLoad(exports, ["getClusterAuth", "getClusterAuthOutput"], () => require("./getClusterAuth"));
utilities.lazyLoad(exports, ["getClusters"], () => require("./getClusters"));
utilities.lazyLoad(exports, ["getNodeGroup", "getNodeGroupOutput"], () => require("./getNodeGroup"));
utilities.lazyLoad(exports, ["getNodeGroups", "getNodeGroupsOutput"], () => require("./getNodeGroups"));
utilities.lazyLoad(exports, ["IdentityProviderConfig"], () => require("./identityProviderConfig"));
utilities.lazyLoad(exports, ["NodeGroup"], () => require("./nodeGroup"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:eks/addon:Addon":
                return new exports.Addon(name, undefined, { urn });
            case "aws:eks/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "aws:eks/fargateProfile:FargateProfile":
                return new exports.FargateProfile(name, undefined, { urn });
            case "aws:eks/identityProviderConfig:IdentityProviderConfig":
                return new exports.IdentityProviderConfig(name, undefined, { urn });
            case "aws:eks/nodeGroup:NodeGroup":
                return new exports.NodeGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "eks/addon", _module);
pulumi.runtime.registerResourceModule("aws", "eks/cluster", _module);
pulumi.runtime.registerResourceModule("aws", "eks/fargateProfile", _module);
pulumi.runtime.registerResourceModule("aws", "eks/identityProviderConfig", _module);
pulumi.runtime.registerResourceModule("aws", "eks/nodeGroup", _module);
