// @pulumi/aws/kms/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicaKey = exports.ReplicaExternalKey = exports.Key = exports.Grant = exports.getSecretsOutput = exports.getSecrets = exports.getSecretOutput = exports.getSecret = exports.getPublicKeyOutput = exports.getPublicKey = exports.getKeyOutput = exports.getKey = exports.getCustomKeyStoreOutput = exports.getCustomKeyStore = exports.getCipherTextOutput = exports.getCipherText = exports.getAliasOutput = exports.getAlias = exports.ExternalKey = exports.CustomKeyStore = exports.Ciphertext = exports.Alias = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Alias = null;
exports.Ciphertext = null;
exports.CustomKeyStore = null;
exports.ExternalKey = null;
exports.getAlias = null;
exports.getAliasOutput = null;
exports.getCipherText = null;
exports.getCipherTextOutput = null;
exports.getCustomKeyStore = null;
exports.getCustomKeyStoreOutput = null;
exports.getKey = null;
exports.getKeyOutput = null;
exports.getPublicKey = null;
exports.getPublicKeyOutput = null;
exports.getSecret = null;
exports.getSecretOutput = null;
exports.getSecrets = null;
exports.getSecretsOutput = null;
exports.Grant = null;
exports.Key = null;
exports.ReplicaExternalKey = null;
exports.ReplicaKey = null;
utilities.lazyLoad(exports, ["Alias"], () => require("./alias"));
utilities.lazyLoad(exports, ["Ciphertext"], () => require("./ciphertext"));
utilities.lazyLoad(exports, ["CustomKeyStore"], () => require("./customKeyStore"));
utilities.lazyLoad(exports, ["ExternalKey"], () => require("./externalKey"));
utilities.lazyLoad(exports, ["getAlias", "getAliasOutput"], () => require("./getAlias"));
utilities.lazyLoad(exports, ["getCipherText", "getCipherTextOutput"], () => require("./getCipherText"));
utilities.lazyLoad(exports, ["getCustomKeyStore", "getCustomKeyStoreOutput"], () => require("./getCustomKeyStore"));
utilities.lazyLoad(exports, ["getKey", "getKeyOutput"], () => require("./getKey"));
utilities.lazyLoad(exports, ["getPublicKey", "getPublicKeyOutput"], () => require("./getPublicKey"));
utilities.lazyLoad(exports, ["getSecret", "getSecretOutput"], () => require("./getSecret"));
utilities.lazyLoad(exports, ["getSecrets", "getSecretsOutput"], () => require("./getSecrets"));
utilities.lazyLoad(exports, ["Grant"], () => require("./grant"));
utilities.lazyLoad(exports, ["Key"], () => require("./key"));
utilities.lazyLoad(exports, ["ReplicaExternalKey"], () => require("./replicaExternalKey"));
utilities.lazyLoad(exports, ["ReplicaKey"], () => require("./replicaKey"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:kms/alias:Alias":
                return new exports.Alias(name, undefined, { urn });
            case "aws:kms/ciphertext:Ciphertext":
                return new exports.Ciphertext(name, undefined, { urn });
            case "aws:kms/customKeyStore:CustomKeyStore":
                return new exports.CustomKeyStore(name, undefined, { urn });
            case "aws:kms/externalKey:ExternalKey":
                return new exports.ExternalKey(name, undefined, { urn });
            case "aws:kms/grant:Grant":
                return new exports.Grant(name, undefined, { urn });
            case "aws:kms/key:Key":
                return new exports.Key(name, undefined, { urn });
            case "aws:kms/replicaExternalKey:ReplicaExternalKey":
                return new exports.ReplicaExternalKey(name, undefined, { urn });
            case "aws:kms/replicaKey:ReplicaKey":
                return new exports.ReplicaKey(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "kms/alias", _module);
pulumi.runtime.registerResourceModule("aws", "kms/ciphertext", _module);
pulumi.runtime.registerResourceModule("aws", "kms/customKeyStore", _module);
pulumi.runtime.registerResourceModule("aws", "kms/externalKey", _module);
pulumi.runtime.registerResourceModule("aws", "kms/grant", _module);
pulumi.runtime.registerResourceModule("aws", "kms/key", _module);
pulumi.runtime.registerResourceModule("aws", "kms/replicaExternalKey", _module);
pulumi.runtime.registerResourceModule("aws", "kms/replicaKey", _module);
