// @pulumi/aws/location/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrackerAssociation = exports.Tracker = exports.RouteCalculation = exports.PlaceIndex = exports.Map = exports.getTrackerAssociationsOutput = exports.getTrackerAssociations = exports.getTrackerAssociationOutput = exports.getTrackerAssociation = exports.getTrackerOutput = exports.getTracker = exports.getRouteCalculatorOutput = exports.getRouteCalculator = exports.getPlaceIndexOutput = exports.getPlaceIndex = exports.getMapOutput = exports.getMap = exports.getGeofenceCollectionOutput = exports.getGeofenceCollection = exports.GeofenceCollection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.GeofenceCollection = null;
exports.getGeofenceCollection = null;
exports.getGeofenceCollectionOutput = null;
exports.getMap = null;
exports.getMapOutput = null;
exports.getPlaceIndex = null;
exports.getPlaceIndexOutput = null;
exports.getRouteCalculator = null;
exports.getRouteCalculatorOutput = null;
exports.getTracker = null;
exports.getTrackerOutput = null;
exports.getTrackerAssociation = null;
exports.getTrackerAssociationOutput = null;
exports.getTrackerAssociations = null;
exports.getTrackerAssociationsOutput = null;
exports.Map = null;
exports.PlaceIndex = null;
exports.RouteCalculation = null;
exports.Tracker = null;
exports.TrackerAssociation = null;
utilities.lazyLoad(exports, ["GeofenceCollection"], () => require("./geofenceCollection"));
utilities.lazyLoad(exports, ["getGeofenceCollection", "getGeofenceCollectionOutput"], () => require("./getGeofenceCollection"));
utilities.lazyLoad(exports, ["getMap", "getMapOutput"], () => require("./getMap"));
utilities.lazyLoad(exports, ["getPlaceIndex", "getPlaceIndexOutput"], () => require("./getPlaceIndex"));
utilities.lazyLoad(exports, ["getRouteCalculator", "getRouteCalculatorOutput"], () => require("./getRouteCalculator"));
utilities.lazyLoad(exports, ["getTracker", "getTrackerOutput"], () => require("./getTracker"));
utilities.lazyLoad(exports, ["getTrackerAssociation", "getTrackerAssociationOutput"], () => require("./getTrackerAssociation"));
utilities.lazyLoad(exports, ["getTrackerAssociations", "getTrackerAssociationsOutput"], () => require("./getTrackerAssociations"));
utilities.lazyLoad(exports, ["Map"], () => require("./map"));
utilities.lazyLoad(exports, ["PlaceIndex"], () => require("./placeIndex"));
utilities.lazyLoad(exports, ["RouteCalculation"], () => require("./routeCalculation"));
utilities.lazyLoad(exports, ["Tracker"], () => require("./tracker"));
utilities.lazyLoad(exports, ["TrackerAssociation"], () => require("./trackerAssociation"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:location/geofenceCollection:GeofenceCollection":
                return new exports.GeofenceCollection(name, undefined, { urn });
            case "aws:location/map:Map":
                return new exports.Map(name, undefined, { urn });
            case "aws:location/placeIndex:PlaceIndex":
                return new exports.PlaceIndex(name, undefined, { urn });
            case "aws:location/routeCalculation:RouteCalculation":
                return new exports.RouteCalculation(name, undefined, { urn });
            case "aws:location/tracker:Tracker":
                return new exports.Tracker(name, undefined, { urn });
            case "aws:location/trackerAssociation:TrackerAssociation":
                return new exports.TrackerAssociation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "location/geofenceCollection", _module);
pulumi.runtime.registerResourceModule("aws", "location/map", _module);
pulumi.runtime.registerResourceModule("aws", "location/placeIndex", _module);
pulumi.runtime.registerResourceModule("aws", "location/routeCalculation", _module);
pulumi.runtime.registerResourceModule("aws", "location/tracker", _module);
pulumi.runtime.registerResourceModule("aws", "location/trackerAssociation", _module);
