// @pulumi/aws/wafv2/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAclLoggingConfiguration = exports.WebAclAssociation = exports.WebAcl = exports.RuleGroup = exports.RegexPatternSet = exports.IpSet = exports.getWebAclOutput = exports.getWebAcl = exports.getRuleGroupOutput = exports.getRuleGroup = exports.getRegexPatternSetOutput = exports.getRegexPatternSet = exports.getIpSetOutput = exports.getIpSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getIpSet = null;
exports.getIpSetOutput = null;
exports.getRegexPatternSet = null;
exports.getRegexPatternSetOutput = null;
exports.getRuleGroup = null;
exports.getRuleGroupOutput = null;
exports.getWebAcl = null;
exports.getWebAclOutput = null;
exports.IpSet = null;
exports.RegexPatternSet = null;
exports.RuleGroup = null;
exports.WebAcl = null;
exports.WebAclAssociation = null;
exports.WebAclLoggingConfiguration = null;
utilities.lazyLoad(exports, ["getIpSet", "getIpSetOutput"], () => require("./getIpSet"));
utilities.lazyLoad(exports, ["getRegexPatternSet", "getRegexPatternSetOutput"], () => require("./getRegexPatternSet"));
utilities.lazyLoad(exports, ["getRuleGroup", "getRuleGroupOutput"], () => require("./getRuleGroup"));
utilities.lazyLoad(exports, ["getWebAcl", "getWebAclOutput"], () => require("./getWebAcl"));
utilities.lazyLoad(exports, ["IpSet"], () => require("./ipSet"));
utilities.lazyLoad(exports, ["RegexPatternSet"], () => require("./regexPatternSet"));
utilities.lazyLoad(exports, ["RuleGroup"], () => require("./ruleGroup"));
utilities.lazyLoad(exports, ["WebAcl"], () => require("./webAcl"));
utilities.lazyLoad(exports, ["WebAclAssociation"], () => require("./webAclAssociation"));
utilities.lazyLoad(exports, ["WebAclLoggingConfiguration"], () => require("./webAclLoggingConfiguration"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:wafv2/ipSet:IpSet":
                return new exports.IpSet(name, undefined, { urn });
            case "aws:wafv2/regexPatternSet:RegexPatternSet":
                return new exports.RegexPatternSet(name, undefined, { urn });
            case "aws:wafv2/ruleGroup:RuleGroup":
                return new exports.RuleGroup(name, undefined, { urn });
            case "aws:wafv2/webAcl:WebAcl":
                return new exports.WebAcl(name, undefined, { urn });
            case "aws:wafv2/webAclAssociation:WebAclAssociation":
                return new exports.WebAclAssociation(name, undefined, { urn });
            case "aws:wafv2/webAclLoggingConfiguration:WebAclLoggingConfiguration":
                return new exports.WebAclLoggingConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "wafv2/ipSet", _module);
pulumi.runtime.registerResourceModule("aws", "wafv2/regexPatternSet", _module);
pulumi.runtime.registerResourceModule("aws", "wafv2/ruleGroup", _module);
pulumi.runtime.registerResourceModule("aws", "wafv2/webAcl", _module);
pulumi.runtime.registerResourceModule("aws", "wafv2/webAclAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "wafv2/webAclLoggingConfiguration", _module);
