// @pulumi/aws/servicecatalog/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagOptionResourceAssociation = exports.TagOption = exports.ServiceAction = exports.ProvisioningArtifact = exports.ProvisionedProduct = exports.ProductPortfolioAssociation = exports.Product = exports.PrincipalPortfolioAssociation = exports.PortfolioShare = exports.Portfolio = exports.OrganizationsAccess = exports.getProductOutput = exports.getProduct = exports.getPortfolioConstraintsOutput = exports.getPortfolioConstraints = exports.getPortfolioOutput = exports.getPortfolio = exports.getLaunchPathsOutput = exports.getLaunchPaths = exports.getConstraintOutput = exports.getConstraint = exports.Constraint = exports.BudgetResourceAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.BudgetResourceAssociation = null;
exports.Constraint = null;
exports.getConstraint = null;
exports.getConstraintOutput = null;
exports.getLaunchPaths = null;
exports.getLaunchPathsOutput = null;
exports.getPortfolio = null;
exports.getPortfolioOutput = null;
exports.getPortfolioConstraints = null;
exports.getPortfolioConstraintsOutput = null;
exports.getProduct = null;
exports.getProductOutput = null;
exports.OrganizationsAccess = null;
exports.Portfolio = null;
exports.PortfolioShare = null;
exports.PrincipalPortfolioAssociation = null;
exports.Product = null;
exports.ProductPortfolioAssociation = null;
exports.ProvisionedProduct = null;
exports.ProvisioningArtifact = null;
exports.ServiceAction = null;
exports.TagOption = null;
exports.TagOptionResourceAssociation = null;
utilities.lazyLoad(exports, ["BudgetResourceAssociation"], () => require("./budgetResourceAssociation"));
utilities.lazyLoad(exports, ["Constraint"], () => require("./constraint"));
utilities.lazyLoad(exports, ["getConstraint", "getConstraintOutput"], () => require("./getConstraint"));
utilities.lazyLoad(exports, ["getLaunchPaths", "getLaunchPathsOutput"], () => require("./getLaunchPaths"));
utilities.lazyLoad(exports, ["getPortfolio", "getPortfolioOutput"], () => require("./getPortfolio"));
utilities.lazyLoad(exports, ["getPortfolioConstraints", "getPortfolioConstraintsOutput"], () => require("./getPortfolioConstraints"));
utilities.lazyLoad(exports, ["getProduct", "getProductOutput"], () => require("./getProduct"));
utilities.lazyLoad(exports, ["OrganizationsAccess"], () => require("./organizationsAccess"));
utilities.lazyLoad(exports, ["Portfolio"], () => require("./portfolio"));
utilities.lazyLoad(exports, ["PortfolioShare"], () => require("./portfolioShare"));
utilities.lazyLoad(exports, ["PrincipalPortfolioAssociation"], () => require("./principalPortfolioAssociation"));
utilities.lazyLoad(exports, ["Product"], () => require("./product"));
utilities.lazyLoad(exports, ["ProductPortfolioAssociation"], () => require("./productPortfolioAssociation"));
utilities.lazyLoad(exports, ["ProvisionedProduct"], () => require("./provisionedProduct"));
utilities.lazyLoad(exports, ["ProvisioningArtifact"], () => require("./provisioningArtifact"));
utilities.lazyLoad(exports, ["ServiceAction"], () => require("./serviceAction"));
utilities.lazyLoad(exports, ["TagOption"], () => require("./tagOption"));
utilities.lazyLoad(exports, ["TagOptionResourceAssociation"], () => require("./tagOptionResourceAssociation"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:servicecatalog/budgetResourceAssociation:BudgetResourceAssociation":
                return new exports.BudgetResourceAssociation(name, undefined, { urn });
            case "aws:servicecatalog/constraint:Constraint":
                return new exports.Constraint(name, undefined, { urn });
            case "aws:servicecatalog/organizationsAccess:OrganizationsAccess":
                return new exports.OrganizationsAccess(name, undefined, { urn });
            case "aws:servicecatalog/portfolio:Portfolio":
                return new exports.Portfolio(name, undefined, { urn });
            case "aws:servicecatalog/portfolioShare:PortfolioShare":
                return new exports.PortfolioShare(name, undefined, { urn });
            case "aws:servicecatalog/principalPortfolioAssociation:PrincipalPortfolioAssociation":
                return new exports.PrincipalPortfolioAssociation(name, undefined, { urn });
            case "aws:servicecatalog/product:Product":
                return new exports.Product(name, undefined, { urn });
            case "aws:servicecatalog/productPortfolioAssociation:ProductPortfolioAssociation":
                return new exports.ProductPortfolioAssociation(name, undefined, { urn });
            case "aws:servicecatalog/provisionedProduct:ProvisionedProduct":
                return new exports.ProvisionedProduct(name, undefined, { urn });
            case "aws:servicecatalog/provisioningArtifact:ProvisioningArtifact":
                return new exports.ProvisioningArtifact(name, undefined, { urn });
            case "aws:servicecatalog/serviceAction:ServiceAction":
                return new exports.ServiceAction(name, undefined, { urn });
            case "aws:servicecatalog/tagOption:TagOption":
                return new exports.TagOption(name, undefined, { urn });
            case "aws:servicecatalog/tagOptionResourceAssociation:TagOptionResourceAssociation":
                return new exports.TagOptionResourceAssociation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "servicecatalog/budgetResourceAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/constraint", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/organizationsAccess", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/portfolio", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/portfolioShare", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/principalPortfolioAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/product", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/productPortfolioAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/provisionedProduct", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/provisioningArtifact", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/serviceAction", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/tagOption", _module);
pulumi.runtime.registerResourceModule("aws", "servicecatalog/tagOptionResourceAssociation", _module);
