// @pulumi/aws/workspaces/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.IpGroup = exports.getWorkspaceOutput = exports.getWorkspace = exports.getImageOutput = exports.getImage = exports.getDirectoryOutput = exports.getDirectory = exports.getBundleOutput = exports.getBundle = exports.Directory = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Directory = null;
exports.getBundle = null;
exports.getBundleOutput = null;
exports.getDirectory = null;
exports.getDirectoryOutput = null;
exports.getImage = null;
exports.getImageOutput = null;
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
exports.IpGroup = null;
exports.Workspace = null;
utilities.lazyLoad(exports, ["Directory"], () => require("./directory"));
utilities.lazyLoad(exports, ["getBundle", "getBundleOutput"], () => require("./getBundle"));
utilities.lazyLoad(exports, ["getDirectory", "getDirectoryOutput"], () => require("./getDirectory"));
utilities.lazyLoad(exports, ["getImage", "getImageOutput"], () => require("./getImage"));
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
utilities.lazyLoad(exports, ["IpGroup"], () => require("./ipGroup"));
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:workspaces/directory:Directory":
                return new exports.Directory(name, undefined, { urn });
            case "aws:workspaces/ipGroup:IpGroup":
                return new exports.IpGroup(name, undefined, { urn });
            case "aws:workspaces/workspace:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "workspaces/directory", _module);
pulumi.runtime.registerResourceModule("aws", "workspaces/ipGroup", _module);
pulumi.runtime.registerResourceModule("aws", "workspaces/workspace", _module);
