// @pulumi/aws/lambda/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisionedConcurrencyConfig = exports.Permission = exports.LayerVersionPermission = exports.LayerVersion = exports.Invocation = exports.getLayerVersionOutput = exports.getLayerVersion = exports.getInvocationOutput = exports.getInvocation = exports.getFunctionUrlOutput = exports.getFunctionUrl = exports.getFunctionOutput = exports.getFunction = exports.getCodeSigningConfigOutput = exports.getCodeSigningConfig = exports.getAliasOutput = exports.getAlias = exports.FunctionUrl = exports.FunctionEventInvokeConfig = exports.Function = exports.EventSourceMapping = exports.CodeSigningConfig = exports.Alias = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Alias = null;
exports.CodeSigningConfig = null;
exports.EventSourceMapping = null;
exports.Function = null;
exports.FunctionEventInvokeConfig = null;
exports.FunctionUrl = null;
exports.getAlias = null;
exports.getAliasOutput = null;
exports.getCodeSigningConfig = null;
exports.getCodeSigningConfigOutput = null;
exports.getFunction = null;
exports.getFunctionOutput = null;
exports.getFunctionUrl = null;
exports.getFunctionUrlOutput = null;
exports.getInvocation = null;
exports.getInvocationOutput = null;
exports.getLayerVersion = null;
exports.getLayerVersionOutput = null;
exports.Invocation = null;
__exportStar(require("./lambdaMixins"), exports);
exports.LayerVersion = null;
exports.LayerVersionPermission = null;
exports.Permission = null;
exports.ProvisionedConcurrencyConfig = null;
__exportStar(require("./runtimes"), exports);
utilities.lazyLoad(exports, ["Alias"], () => require("./alias"));
utilities.lazyLoad(exports, ["CodeSigningConfig"], () => require("./codeSigningConfig"));
utilities.lazyLoad(exports, ["EventSourceMapping"], () => require("./eventSourceMapping"));
utilities.lazyLoad(exports, ["Function"], () => require("./function"));
utilities.lazyLoad(exports, ["FunctionEventInvokeConfig"], () => require("./functionEventInvokeConfig"));
utilities.lazyLoad(exports, ["FunctionUrl"], () => require("./functionUrl"));
utilities.lazyLoad(exports, ["getAlias", "getAliasOutput"], () => require("./getAlias"));
utilities.lazyLoad(exports, ["getCodeSigningConfig", "getCodeSigningConfigOutput"], () => require("./getCodeSigningConfig"));
utilities.lazyLoad(exports, ["getFunction", "getFunctionOutput"], () => require("./getFunction"));
utilities.lazyLoad(exports, ["getFunctionUrl", "getFunctionUrlOutput"], () => require("./getFunctionUrl"));
utilities.lazyLoad(exports, ["getInvocation", "getInvocationOutput"], () => require("./getInvocation"));
utilities.lazyLoad(exports, ["getLayerVersion", "getLayerVersionOutput"], () => require("./getLayerVersion"));
utilities.lazyLoad(exports, ["Invocation"], () => require("./invocation"));
utilities.lazyLoad(exports, ["LayerVersion"], () => require("./layerVersion"));
utilities.lazyLoad(exports, ["LayerVersionPermission"], () => require("./layerVersionPermission"));
utilities.lazyLoad(exports, ["Permission"], () => require("./permission"));
utilities.lazyLoad(exports, ["ProvisionedConcurrencyConfig"], () => require("./provisionedConcurrencyConfig"));
// Export enums:
__exportStar(require("../types/enums/lambda"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:lambda/alias:Alias":
                return new exports.Alias(name, undefined, { urn });
            case "aws:lambda/codeSigningConfig:CodeSigningConfig":
                return new exports.CodeSigningConfig(name, undefined, { urn });
            case "aws:lambda/eventSourceMapping:EventSourceMapping":
                return new exports.EventSourceMapping(name, undefined, { urn });
            case "aws:lambda/function:Function":
                return new exports.Function(name, undefined, { urn });
            case "aws:lambda/functionEventInvokeConfig:FunctionEventInvokeConfig":
                return new exports.FunctionEventInvokeConfig(name, undefined, { urn });
            case "aws:lambda/functionUrl:FunctionUrl":
                return new exports.FunctionUrl(name, undefined, { urn });
            case "aws:lambda/invocation:Invocation":
                return new exports.Invocation(name, undefined, { urn });
            case "aws:lambda/layerVersion:LayerVersion":
                return new exports.LayerVersion(name, undefined, { urn });
            case "aws:lambda/layerVersionPermission:LayerVersionPermission":
                return new exports.LayerVersionPermission(name, undefined, { urn });
            case "aws:lambda/permission:Permission":
                return new exports.Permission(name, undefined, { urn });
            case "aws:lambda/provisionedConcurrencyConfig:ProvisionedConcurrencyConfig":
                return new exports.ProvisionedConcurrencyConfig(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "lambda/alias", _module);
pulumi.runtime.registerResourceModule("aws", "lambda/codeSigningConfig", _module);
pulumi.runtime.registerResourceModule("aws", "lambda/eventSourceMapping", _module);
pulumi.runtime.registerResourceModule("aws", "lambda/function", _module);
pulumi.runtime.registerResourceModule("aws", "lambda/functionEventInvokeConfig", _module);
pulumi.runtime.registerResourceModule("aws", "lambda/functionUrl", _module);
pulumi.runtime.registerResourceModule("aws", "lambda/invocation", _module);
pulumi.runtime.registerResourceModule("aws", "lambda/layerVersion", _module);
pulumi.runtime.registerResourceModule("aws", "lambda/layerVersionPermission", _module);
pulumi.runtime.registerResourceModule("aws", "lambda/permission", _module);
pulumi.runtime.registerResourceModule("aws", "lambda/provisionedConcurrencyConfig", _module);
