// @pulumi/aws/dynamodb/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tag = exports.TableReplica = exports.TableItem = exports.Table = exports.KinesisStreamingDestination = exports.GlobalTable = exports.getTableOutput = exports.getTable = exports.ContributorInsights = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ContributorInsights = null;
__exportStar(require("./dynamodbMixins"), exports);
exports.getTable = null;
exports.getTableOutput = null;
exports.GlobalTable = null;
exports.KinesisStreamingDestination = null;
exports.Table = null;
exports.TableItem = null;
exports.TableReplica = null;
exports.Tag = null;
utilities.lazyLoad(exports, ["ContributorInsights"], () => require("./contributorInsights"));
utilities.lazyLoad(exports, ["getTable", "getTableOutput"], () => require("./getTable"));
utilities.lazyLoad(exports, ["GlobalTable"], () => require("./globalTable"));
utilities.lazyLoad(exports, ["KinesisStreamingDestination"], () => require("./kinesisStreamingDestination"));
utilities.lazyLoad(exports, ["Table"], () => require("./table"));
utilities.lazyLoad(exports, ["TableItem"], () => require("./tableItem"));
utilities.lazyLoad(exports, ["TableReplica"], () => require("./tableReplica"));
utilities.lazyLoad(exports, ["Tag"], () => require("./tag"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:dynamodb/contributorInsights:ContributorInsights":
                return new exports.ContributorInsights(name, undefined, { urn });
            case "aws:dynamodb/globalTable:GlobalTable":
                return new exports.GlobalTable(name, undefined, { urn });
            case "aws:dynamodb/kinesisStreamingDestination:KinesisStreamingDestination":
                return new exports.KinesisStreamingDestination(name, undefined, { urn });
            case "aws:dynamodb/table:Table":
                return new exports.Table(name, undefined, { urn });
            case "aws:dynamodb/tableItem:TableItem":
                return new exports.TableItem(name, undefined, { urn });
            case "aws:dynamodb/tableReplica:TableReplica":
                return new exports.TableReplica(name, undefined, { urn });
            case "aws:dynamodb/tag:Tag":
                return new exports.Tag(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "dynamodb/contributorInsights", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/globalTable", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/kinesisStreamingDestination", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/table", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/tableItem", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/tableReplica", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/tag", _module);
