// @pulumi/aws/networkfirewall/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleGroup = exports.ResourcePolicy = exports.LoggingConfiguration = exports.getFirewallPolicyOutput = exports.getFirewallPolicy = exports.getFirewallOutput = exports.getFirewall = exports.FirewallPolicy = exports.Firewall = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Firewall = null;
exports.FirewallPolicy = null;
exports.getFirewall = null;
exports.getFirewallOutput = null;
exports.getFirewallPolicy = null;
exports.getFirewallPolicyOutput = null;
exports.LoggingConfiguration = null;
exports.ResourcePolicy = null;
exports.RuleGroup = null;
utilities.lazyLoad(exports, ["Firewall"], () => require("./firewall"));
utilities.lazyLoad(exports, ["FirewallPolicy"], () => require("./firewallPolicy"));
utilities.lazyLoad(exports, ["getFirewall", "getFirewallOutput"], () => require("./getFirewall"));
utilities.lazyLoad(exports, ["getFirewallPolicy", "getFirewallPolicyOutput"], () => require("./getFirewallPolicy"));
utilities.lazyLoad(exports, ["LoggingConfiguration"], () => require("./loggingConfiguration"));
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
utilities.lazyLoad(exports, ["RuleGroup"], () => require("./ruleGroup"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:networkfirewall/firewall:Firewall":
                return new exports.Firewall(name, undefined, { urn });
            case "aws:networkfirewall/firewallPolicy:FirewallPolicy":
                return new exports.FirewallPolicy(name, undefined, { urn });
            case "aws:networkfirewall/loggingConfiguration:LoggingConfiguration":
                return new exports.LoggingConfiguration(name, undefined, { urn });
            case "aws:networkfirewall/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws:networkfirewall/ruleGroup:RuleGroup":
                return new exports.RuleGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "networkfirewall/firewall", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/firewallPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/loggingConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/ruleGroup", _module);
