// @pulumi/aws/batch/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchedulingPolicy = exports.JobQueue = exports.JobDefinition = exports.getSchedulingPolicyOutput = exports.getSchedulingPolicy = exports.getJobQueueOutput = exports.getJobQueue = exports.getComputeEnvironmentOutput = exports.getComputeEnvironment = exports.ComputeEnvironment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ComputeEnvironment = null;
exports.getComputeEnvironment = null;
exports.getComputeEnvironmentOutput = null;
exports.getJobQueue = null;
exports.getJobQueueOutput = null;
exports.getSchedulingPolicy = null;
exports.getSchedulingPolicyOutput = null;
exports.JobDefinition = null;
exports.JobQueue = null;
exports.SchedulingPolicy = null;
utilities.lazyLoad(exports, ["ComputeEnvironment"], () => require("./computeEnvironment"));
utilities.lazyLoad(exports, ["getComputeEnvironment", "getComputeEnvironmentOutput"], () => require("./getComputeEnvironment"));
utilities.lazyLoad(exports, ["getJobQueue", "getJobQueueOutput"], () => require("./getJobQueue"));
utilities.lazyLoad(exports, ["getSchedulingPolicy", "getSchedulingPolicyOutput"], () => require("./getSchedulingPolicy"));
utilities.lazyLoad(exports, ["JobDefinition"], () => require("./jobDefinition"));
utilities.lazyLoad(exports, ["JobQueue"], () => require("./jobQueue"));
utilities.lazyLoad(exports, ["SchedulingPolicy"], () => require("./schedulingPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:batch/computeEnvironment:ComputeEnvironment":
                return new exports.ComputeEnvironment(name, undefined, { urn });
            case "aws:batch/jobDefinition:JobDefinition":
                return new exports.JobDefinition(name, undefined, { urn });
            case "aws:batch/jobQueue:JobQueue":
                return new exports.JobQueue(name, undefined, { urn });
            case "aws:batch/schedulingPolicy:SchedulingPolicy":
                return new exports.SchedulingPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "batch/computeEnvironment", _module);
pulumi.runtime.registerResourceModule("aws", "batch/jobDefinition", _module);
pulumi.runtime.registerResourceModule("aws", "batch/jobQueue", _module);
pulumi.runtime.registerResourceModule("aws", "batch/schedulingPolicy", _module);
