// @pulumi/aws/ecs/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskSet = exports.TaskDefinition = exports.Tag = exports.Service = exports.getTaskDefinitionOutput = exports.getTaskDefinition = exports.getServiceOutput = exports.getService = exports.getContainerDefinitionOutput = exports.getContainerDefinition = exports.getClusterOutput = exports.getCluster = exports.ClusterCapacityProviders = exports.Cluster = exports.CapacityProvider = exports.AccountSettingDefault = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccountSettingDefault = null;
exports.CapacityProvider = null;
exports.Cluster = null;
exports.ClusterCapacityProviders = null;
__exportStar(require("./container"), exports);
exports.getCluster = null;
exports.getClusterOutput = null;
exports.getContainerDefinition = null;
exports.getContainerDefinitionOutput = null;
exports.getService = null;
exports.getServiceOutput = null;
exports.getTaskDefinition = null;
exports.getTaskDefinitionOutput = null;
exports.Service = null;
exports.Tag = null;
exports.TaskDefinition = null;
exports.TaskSet = null;
utilities.lazyLoad(exports, ["AccountSettingDefault"], () => require("./accountSettingDefault"));
utilities.lazyLoad(exports, ["CapacityProvider"], () => require("./capacityProvider"));
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
utilities.lazyLoad(exports, ["ClusterCapacityProviders"], () => require("./clusterCapacityProviders"));
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
utilities.lazyLoad(exports, ["getContainerDefinition", "getContainerDefinitionOutput"], () => require("./getContainerDefinition"));
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
utilities.lazyLoad(exports, ["getTaskDefinition", "getTaskDefinitionOutput"], () => require("./getTaskDefinition"));
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
utilities.lazyLoad(exports, ["Tag"], () => require("./tag"));
utilities.lazyLoad(exports, ["TaskDefinition"], () => require("./taskDefinition"));
utilities.lazyLoad(exports, ["TaskSet"], () => require("./taskSet"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:ecs/accountSettingDefault:AccountSettingDefault":
                return new exports.AccountSettingDefault(name, undefined, { urn });
            case "aws:ecs/capacityProvider:CapacityProvider":
                return new exports.CapacityProvider(name, undefined, { urn });
            case "aws:ecs/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "aws:ecs/clusterCapacityProviders:ClusterCapacityProviders":
                return new exports.ClusterCapacityProviders(name, undefined, { urn });
            case "aws:ecs/service:Service":
                return new exports.Service(name, undefined, { urn });
            case "aws:ecs/tag:Tag":
                return new exports.Tag(name, undefined, { urn });
            case "aws:ecs/taskDefinition:TaskDefinition":
                return new exports.TaskDefinition(name, undefined, { urn });
            case "aws:ecs/taskSet:TaskSet":
                return new exports.TaskSet(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "ecs/accountSettingDefault", _module);
pulumi.runtime.registerResourceModule("aws", "ecs/capacityProvider", _module);
pulumi.runtime.registerResourceModule("aws", "ecs/cluster", _module);
pulumi.runtime.registerResourceModule("aws", "ecs/clusterCapacityProviders", _module);
pulumi.runtime.registerResourceModule("aws", "ecs/service", _module);
pulumi.runtime.registerResourceModule("aws", "ecs/tag", _module);
pulumi.runtime.registerResourceModule("aws", "ecs/taskDefinition", _module);
pulumi.runtime.registerResourceModule("aws", "ecs/taskSet", _module);
