// @pulumi/aws/s3/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectCopy = exports.Inventory = exports.getObjectsOutput = exports.getObjects = exports.getObjectOutput = exports.getObject = exports.getCanonicalUserId = exports.getBucketPolicyOutput = exports.getBucketPolicy = exports.getBucketObjectsOutput = exports.getBucketObjects = exports.getBucketObjectOutput = exports.getBucketObject = exports.getBucketOutput = exports.getBucket = exports.getAccountPublicAccessBlockOutput = exports.getAccountPublicAccessBlock = exports.BucketWebsiteConfigurationV2 = exports.BucketVersioningV2 = exports.BucketV2 = exports.BucketServerSideEncryptionConfigurationV2 = exports.BucketRequestPaymentConfigurationV2 = exports.BucketReplicationConfig = exports.BucketPublicAccessBlock = exports.BucketPolicy = exports.BucketOwnershipControls = exports.BucketObjectv2 = exports.BucketObjectLockConfigurationV2 = exports.BucketObject = exports.BucketNotification = exports.BucketMetric = exports.BucketLoggingV2 = exports.BucketLifecycleConfigurationV2 = exports.BucketIntelligentTieringConfiguration = exports.BucketCorsConfigurationV2 = exports.BucketAclV2 = exports.BucketAccelerateConfigurationV2 = exports.Bucket = exports.AnalyticsConfiguration = exports.AccountPublicAccessBlock = exports.AccessPoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccessPoint = null;
exports.AccountPublicAccessBlock = null;
exports.AnalyticsConfiguration = null;
exports.Bucket = null;
exports.BucketAccelerateConfigurationV2 = null;
exports.BucketAclV2 = null;
exports.BucketCorsConfigurationV2 = null;
exports.BucketIntelligentTieringConfiguration = null;
exports.BucketLifecycleConfigurationV2 = null;
exports.BucketLoggingV2 = null;
exports.BucketMetric = null;
exports.BucketNotification = null;
exports.BucketObject = null;
exports.BucketObjectLockConfigurationV2 = null;
exports.BucketObjectv2 = null;
exports.BucketOwnershipControls = null;
exports.BucketPolicy = null;
exports.BucketPublicAccessBlock = null;
exports.BucketReplicationConfig = null;
exports.BucketRequestPaymentConfigurationV2 = null;
exports.BucketServerSideEncryptionConfigurationV2 = null;
exports.BucketV2 = null;
exports.BucketVersioningV2 = null;
exports.BucketWebsiteConfigurationV2 = null;
__exportStar(require("./cannedAcl"), exports);
exports.getAccountPublicAccessBlock = null;
exports.getAccountPublicAccessBlockOutput = null;
exports.getBucket = null;
exports.getBucketOutput = null;
exports.getBucketObject = null;
exports.getBucketObjectOutput = null;
exports.getBucketObjects = null;
exports.getBucketObjectsOutput = null;
exports.getBucketPolicy = null;
exports.getBucketPolicyOutput = null;
exports.getCanonicalUserId = null;
exports.getObject = null;
exports.getObjectOutput = null;
exports.getObjects = null;
exports.getObjectsOutput = null;
exports.Inventory = null;
exports.ObjectCopy = null;
__exportStar(require("./routingRules"), exports);
__exportStar(require("./s3Mixins"), exports);
utilities.lazyLoad(exports, ["AccessPoint"], () => require("./accessPoint"));
utilities.lazyLoad(exports, ["AccountPublicAccessBlock"], () => require("./accountPublicAccessBlock"));
utilities.lazyLoad(exports, ["AnalyticsConfiguration"], () => require("./analyticsConfiguration"));
utilities.lazyLoad(exports, ["Bucket"], () => require("./bucket"));
utilities.lazyLoad(exports, ["BucketAccelerateConfigurationV2"], () => require("./bucketAccelerateConfigurationV2"));
utilities.lazyLoad(exports, ["BucketAclV2"], () => require("./bucketAclV2"));
utilities.lazyLoad(exports, ["BucketCorsConfigurationV2"], () => require("./bucketCorsConfigurationV2"));
utilities.lazyLoad(exports, ["BucketIntelligentTieringConfiguration"], () => require("./bucketIntelligentTieringConfiguration"));
utilities.lazyLoad(exports, ["BucketLifecycleConfigurationV2"], () => require("./bucketLifecycleConfigurationV2"));
utilities.lazyLoad(exports, ["BucketLoggingV2"], () => require("./bucketLoggingV2"));
utilities.lazyLoad(exports, ["BucketMetric"], () => require("./bucketMetric"));
utilities.lazyLoad(exports, ["BucketNotification"], () => require("./bucketNotification"));
utilities.lazyLoad(exports, ["BucketObject"], () => require("./bucketObject"));
utilities.lazyLoad(exports, ["BucketObjectLockConfigurationV2"], () => require("./bucketObjectLockConfigurationV2"));
utilities.lazyLoad(exports, ["BucketObjectv2"], () => require("./bucketObjectv2"));
utilities.lazyLoad(exports, ["BucketOwnershipControls"], () => require("./bucketOwnershipControls"));
utilities.lazyLoad(exports, ["BucketPolicy"], () => require("./bucketPolicy"));
utilities.lazyLoad(exports, ["BucketPublicAccessBlock"], () => require("./bucketPublicAccessBlock"));
utilities.lazyLoad(exports, ["BucketReplicationConfig"], () => require("./bucketReplicationConfig"));
utilities.lazyLoad(exports, ["BucketRequestPaymentConfigurationV2"], () => require("./bucketRequestPaymentConfigurationV2"));
utilities.lazyLoad(exports, ["BucketServerSideEncryptionConfigurationV2"], () => require("./bucketServerSideEncryptionConfigurationV2"));
utilities.lazyLoad(exports, ["BucketV2"], () => require("./bucketV2"));
utilities.lazyLoad(exports, ["BucketVersioningV2"], () => require("./bucketVersioningV2"));
utilities.lazyLoad(exports, ["BucketWebsiteConfigurationV2"], () => require("./bucketWebsiteConfigurationV2"));
utilities.lazyLoad(exports, ["getAccountPublicAccessBlock", "getAccountPublicAccessBlockOutput"], () => require("./getAccountPublicAccessBlock"));
utilities.lazyLoad(exports, ["getBucket", "getBucketOutput"], () => require("./getBucket"));
utilities.lazyLoad(exports, ["getBucketObject", "getBucketObjectOutput"], () => require("./getBucketObject"));
utilities.lazyLoad(exports, ["getBucketObjects", "getBucketObjectsOutput"], () => require("./getBucketObjects"));
utilities.lazyLoad(exports, ["getBucketPolicy", "getBucketPolicyOutput"], () => require("./getBucketPolicy"));
utilities.lazyLoad(exports, ["getCanonicalUserId"], () => require("./getCanonicalUserId"));
utilities.lazyLoad(exports, ["getObject", "getObjectOutput"], () => require("./getObject"));
utilities.lazyLoad(exports, ["getObjects", "getObjectsOutput"], () => require("./getObjects"));
utilities.lazyLoad(exports, ["Inventory"], () => require("./inventory"));
utilities.lazyLoad(exports, ["ObjectCopy"], () => require("./objectCopy"));
// Export enums:
__exportStar(require("../types/enums/s3"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:s3/accessPoint:AccessPoint":
                return new exports.AccessPoint(name, undefined, { urn });
            case "aws:s3/accountPublicAccessBlock:AccountPublicAccessBlock":
                return new exports.AccountPublicAccessBlock(name, undefined, { urn });
            case "aws:s3/analyticsConfiguration:AnalyticsConfiguration":
                return new exports.AnalyticsConfiguration(name, undefined, { urn });
            case "aws:s3/bucket:Bucket":
                return new exports.Bucket(name, undefined, { urn });
            case "aws:s3/bucketAccelerateConfigurationV2:BucketAccelerateConfigurationV2":
                return new exports.BucketAccelerateConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketAclV2:BucketAclV2":
                return new exports.BucketAclV2(name, undefined, { urn });
            case "aws:s3/bucketCorsConfigurationV2:BucketCorsConfigurationV2":
                return new exports.BucketCorsConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketIntelligentTieringConfiguration:BucketIntelligentTieringConfiguration":
                return new exports.BucketIntelligentTieringConfiguration(name, undefined, { urn });
            case "aws:s3/bucketLifecycleConfigurationV2:BucketLifecycleConfigurationV2":
                return new exports.BucketLifecycleConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketLoggingV2:BucketLoggingV2":
                return new exports.BucketLoggingV2(name, undefined, { urn });
            case "aws:s3/bucketMetric:BucketMetric":
                return new exports.BucketMetric(name, undefined, { urn });
            case "aws:s3/bucketNotification:BucketNotification":
                return new exports.BucketNotification(name, undefined, { urn });
            case "aws:s3/bucketObject:BucketObject":
                return new exports.BucketObject(name, undefined, { urn });
            case "aws:s3/bucketObjectLockConfigurationV2:BucketObjectLockConfigurationV2":
                return new exports.BucketObjectLockConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketObjectv2:BucketObjectv2":
                return new exports.BucketObjectv2(name, undefined, { urn });
            case "aws:s3/bucketOwnershipControls:BucketOwnershipControls":
                return new exports.BucketOwnershipControls(name, undefined, { urn });
            case "aws:s3/bucketPolicy:BucketPolicy":
                return new exports.BucketPolicy(name, undefined, { urn });
            case "aws:s3/bucketPublicAccessBlock:BucketPublicAccessBlock":
                return new exports.BucketPublicAccessBlock(name, undefined, { urn });
            case "aws:s3/bucketReplicationConfig:BucketReplicationConfig":
                return new exports.BucketReplicationConfig(name, undefined, { urn });
            case "aws:s3/bucketRequestPaymentConfigurationV2:BucketRequestPaymentConfigurationV2":
                return new exports.BucketRequestPaymentConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketServerSideEncryptionConfigurationV2:BucketServerSideEncryptionConfigurationV2":
                return new exports.BucketServerSideEncryptionConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketV2:BucketV2":
                return new exports.BucketV2(name, undefined, { urn });
            case "aws:s3/bucketVersioningV2:BucketVersioningV2":
                return new exports.BucketVersioningV2(name, undefined, { urn });
            case "aws:s3/bucketWebsiteConfigurationV2:BucketWebsiteConfigurationV2":
                return new exports.BucketWebsiteConfigurationV2(name, undefined, { urn });
            case "aws:s3/inventory:Inventory":
                return new exports.Inventory(name, undefined, { urn });
            case "aws:s3/objectCopy:ObjectCopy":
                return new exports.ObjectCopy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "s3/accessPoint", _module);
pulumi.runtime.registerResourceModule("aws", "s3/accountPublicAccessBlock", _module);
pulumi.runtime.registerResourceModule("aws", "s3/analyticsConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucket", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketAccelerateConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketAclV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketCorsConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketIntelligentTieringConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketLifecycleConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketLoggingV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketMetric", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketNotification", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketObject", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketObjectLockConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketObjectv2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketOwnershipControls", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketPublicAccessBlock", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketReplicationConfig", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketRequestPaymentConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketServerSideEncryptionConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketVersioningV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketWebsiteConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/inventory", _module);
pulumi.runtime.registerResourceModule("aws", "s3/objectCopy", _module);
