// @pulumi/aws/appstream/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserStackAssociation = exports.User = exports.Stack = exports.ImageBuilder = exports.FleetStackAssociation = exports.Fleet = exports.DirectoryConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DirectoryConfig = null;
exports.Fleet = null;
exports.FleetStackAssociation = null;
exports.ImageBuilder = null;
exports.Stack = null;
exports.User = null;
exports.UserStackAssociation = null;
utilities.lazyLoad(exports, ["DirectoryConfig"], () => require("./directoryConfig"));
utilities.lazyLoad(exports, ["Fleet"], () => require("./fleet"));
utilities.lazyLoad(exports, ["FleetStackAssociation"], () => require("./fleetStackAssociation"));
utilities.lazyLoad(exports, ["ImageBuilder"], () => require("./imageBuilder"));
utilities.lazyLoad(exports, ["Stack"], () => require("./stack"));
utilities.lazyLoad(exports, ["User"], () => require("./user"));
utilities.lazyLoad(exports, ["UserStackAssociation"], () => require("./userStackAssociation"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:appstream/directoryConfig:DirectoryConfig":
                return new exports.DirectoryConfig(name, undefined, { urn });
            case "aws:appstream/fleet:Fleet":
                return new exports.Fleet(name, undefined, { urn });
            case "aws:appstream/fleetStackAssociation:FleetStackAssociation":
                return new exports.FleetStackAssociation(name, undefined, { urn });
            case "aws:appstream/imageBuilder:ImageBuilder":
                return new exports.ImageBuilder(name, undefined, { urn });
            case "aws:appstream/stack:Stack":
                return new exports.Stack(name, undefined, { urn });
            case "aws:appstream/user:User":
                return new exports.User(name, undefined, { urn });
            case "aws:appstream/userStackAssociation:UserStackAssociation":
                return new exports.UserStackAssociation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "appstream/directoryConfig", _module);
pulumi.runtime.registerResourceModule("aws", "appstream/fleet", _module);
pulumi.runtime.registerResourceModule("aws", "appstream/fleetStackAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "appstream/imageBuilder", _module);
pulumi.runtime.registerResourceModule("aws", "appstream/stack", _module);
pulumi.runtime.registerResourceModule("aws", "appstream/user", _module);
pulumi.runtime.registerResourceModule("aws", "appstream/userStackAssociation", _module);
