// @pulumi/aws/cfg/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Rule = exports.RemediationConfiguration = exports.RecorderStatus = exports.Recorder = exports.OrganizationManagedRule = exports.OrganizationCustomRule = exports.OrganizationConformancePack = exports.DeliveryChannel = exports.ConformancePack = exports.ConfigurationAggregator = exports.AggregateAuthorization = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AggregateAuthorization = null;
exports.ConfigurationAggregator = null;
exports.ConformancePack = null;
exports.DeliveryChannel = null;
exports.OrganizationConformancePack = null;
exports.OrganizationCustomRule = null;
exports.OrganizationManagedRule = null;
exports.Recorder = null;
exports.RecorderStatus = null;
exports.RemediationConfiguration = null;
exports.Rule = null;
utilities.lazyLoad(exports, ["AggregateAuthorization"], () => require("./aggregateAuthorization"));
utilities.lazyLoad(exports, ["ConfigurationAggregator"], () => require("./configurationAggregator"));
utilities.lazyLoad(exports, ["ConformancePack"], () => require("./conformancePack"));
utilities.lazyLoad(exports, ["DeliveryChannel"], () => require("./deliveryChannel"));
utilities.lazyLoad(exports, ["OrganizationConformancePack"], () => require("./organizationConformancePack"));
utilities.lazyLoad(exports, ["OrganizationCustomRule"], () => require("./organizationCustomRule"));
utilities.lazyLoad(exports, ["OrganizationManagedRule"], () => require("./organizationManagedRule"));
utilities.lazyLoad(exports, ["Recorder"], () => require("./recorder"));
utilities.lazyLoad(exports, ["RecorderStatus"], () => require("./recorderStatus"));
utilities.lazyLoad(exports, ["RemediationConfiguration"], () => require("./remediationConfiguration"));
utilities.lazyLoad(exports, ["Rule"], () => require("./rule"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:cfg/aggregateAuthorization:AggregateAuthorization":
                return new exports.AggregateAuthorization(name, undefined, { urn });
            case "aws:cfg/configurationAggregator:ConfigurationAggregator":
                return new exports.ConfigurationAggregator(name, undefined, { urn });
            case "aws:cfg/conformancePack:ConformancePack":
                return new exports.ConformancePack(name, undefined, { urn });
            case "aws:cfg/deliveryChannel:DeliveryChannel":
                return new exports.DeliveryChannel(name, undefined, { urn });
            case "aws:cfg/organizationConformancePack:OrganizationConformancePack":
                return new exports.OrganizationConformancePack(name, undefined, { urn });
            case "aws:cfg/organizationCustomRule:OrganizationCustomRule":
                return new exports.OrganizationCustomRule(name, undefined, { urn });
            case "aws:cfg/organizationManagedRule:OrganizationManagedRule":
                return new exports.OrganizationManagedRule(name, undefined, { urn });
            case "aws:cfg/recorder:Recorder":
                return new exports.Recorder(name, undefined, { urn });
            case "aws:cfg/recorderStatus:RecorderStatus":
                return new exports.RecorderStatus(name, undefined, { urn });
            case "aws:cfg/remediationConfiguration:RemediationConfiguration":
                return new exports.RemediationConfiguration(name, undefined, { urn });
            case "aws:cfg/rule:Rule":
                return new exports.Rule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "cfg/aggregateAuthorization", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/configurationAggregator", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/conformancePack", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/deliveryChannel", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/organizationConformancePack", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/organizationCustomRule", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/organizationManagedRule", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/recorder", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/recorderStatus", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/remediationConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/rule", _module);
