// @pulumi/aws/fsx/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsFileSystem = exports.OpenZfsVolume = exports.OpenZfsSnapshot = exports.OpenZfsFileSystem = exports.OntapVolume = exports.OntapStorageVirtualMachine = exports.OntapFileSystem = exports.LustreFileSystem = exports.getOpenZfsSnapshotOutput = exports.getOpenZfsSnapshot = exports.DataRepositoryAssociation = exports.Backup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Backup = null;
exports.DataRepositoryAssociation = null;
exports.getOpenZfsSnapshot = null;
exports.getOpenZfsSnapshotOutput = null;
exports.LustreFileSystem = null;
exports.OntapFileSystem = null;
exports.OntapStorageVirtualMachine = null;
exports.OntapVolume = null;
exports.OpenZfsFileSystem = null;
exports.OpenZfsSnapshot = null;
exports.OpenZfsVolume = null;
exports.WindowsFileSystem = null;
utilities.lazyLoad(exports, ["Backup"], () => require("./backup"));
utilities.lazyLoad(exports, ["DataRepositoryAssociation"], () => require("./dataRepositoryAssociation"));
utilities.lazyLoad(exports, ["getOpenZfsSnapshot", "getOpenZfsSnapshotOutput"], () => require("./getOpenZfsSnapshot"));
utilities.lazyLoad(exports, ["LustreFileSystem"], () => require("./lustreFileSystem"));
utilities.lazyLoad(exports, ["OntapFileSystem"], () => require("./ontapFileSystem"));
utilities.lazyLoad(exports, ["OntapStorageVirtualMachine"], () => require("./ontapStorageVirtualMachine"));
utilities.lazyLoad(exports, ["OntapVolume"], () => require("./ontapVolume"));
utilities.lazyLoad(exports, ["OpenZfsFileSystem"], () => require("./openZfsFileSystem"));
utilities.lazyLoad(exports, ["OpenZfsSnapshot"], () => require("./openZfsSnapshot"));
utilities.lazyLoad(exports, ["OpenZfsVolume"], () => require("./openZfsVolume"));
utilities.lazyLoad(exports, ["WindowsFileSystem"], () => require("./windowsFileSystem"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:fsx/backup:Backup":
                return new exports.Backup(name, undefined, { urn });
            case "aws:fsx/dataRepositoryAssociation:DataRepositoryAssociation":
                return new exports.DataRepositoryAssociation(name, undefined, { urn });
            case "aws:fsx/lustreFileSystem:LustreFileSystem":
                return new exports.LustreFileSystem(name, undefined, { urn });
            case "aws:fsx/ontapFileSystem:OntapFileSystem":
                return new exports.OntapFileSystem(name, undefined, { urn });
            case "aws:fsx/ontapStorageVirtualMachine:OntapStorageVirtualMachine":
                return new exports.OntapStorageVirtualMachine(name, undefined, { urn });
            case "aws:fsx/ontapVolume:OntapVolume":
                return new exports.OntapVolume(name, undefined, { urn });
            case "aws:fsx/openZfsFileSystem:OpenZfsFileSystem":
                return new exports.OpenZfsFileSystem(name, undefined, { urn });
            case "aws:fsx/openZfsSnapshot:OpenZfsSnapshot":
                return new exports.OpenZfsSnapshot(name, undefined, { urn });
            case "aws:fsx/openZfsVolume:OpenZfsVolume":
                return new exports.OpenZfsVolume(name, undefined, { urn });
            case "aws:fsx/windowsFileSystem:WindowsFileSystem":
                return new exports.WindowsFileSystem(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "fsx/backup", _module);
pulumi.runtime.registerResourceModule("aws", "fsx/dataRepositoryAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "fsx/lustreFileSystem", _module);
pulumi.runtime.registerResourceModule("aws", "fsx/ontapFileSystem", _module);
pulumi.runtime.registerResourceModule("aws", "fsx/ontapStorageVirtualMachine", _module);
pulumi.runtime.registerResourceModule("aws", "fsx/ontapVolume", _module);
pulumi.runtime.registerResourceModule("aws", "fsx/openZfsFileSystem", _module);
pulumi.runtime.registerResourceModule("aws", "fsx/openZfsSnapshot", _module);
pulumi.runtime.registerResourceModule("aws", "fsx/openZfsVolume", _module);
pulumi.runtime.registerResourceModule("aws", "fsx/windowsFileSystem", _module);
