// @pulumi/aws/kinesis/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VideoStream = exports.StreamConsumer = exports.Stream = exports.getStreamConsumerOutput = exports.getStreamConsumer = exports.getStreamOutput = exports.getStream = exports.getFirehoseDeliveryStreamOutput = exports.getFirehoseDeliveryStream = exports.FirehoseDeliveryStream = exports.AnalyticsApplication = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AnalyticsApplication = null;
exports.FirehoseDeliveryStream = null;
exports.getFirehoseDeliveryStream = null;
exports.getFirehoseDeliveryStreamOutput = null;
exports.getStream = null;
exports.getStreamOutput = null;
exports.getStreamConsumer = null;
exports.getStreamConsumerOutput = null;
__exportStar(require("./kinesisMixins"), exports);
exports.Stream = null;
exports.StreamConsumer = null;
exports.VideoStream = null;
utilities.lazyLoad(exports, ["AnalyticsApplication"], () => require("./analyticsApplication"));
utilities.lazyLoad(exports, ["FirehoseDeliveryStream"], () => require("./firehoseDeliveryStream"));
utilities.lazyLoad(exports, ["getFirehoseDeliveryStream", "getFirehoseDeliveryStreamOutput"], () => require("./getFirehoseDeliveryStream"));
utilities.lazyLoad(exports, ["getStream", "getStreamOutput"], () => require("./getStream"));
utilities.lazyLoad(exports, ["getStreamConsumer", "getStreamConsumerOutput"], () => require("./getStreamConsumer"));
utilities.lazyLoad(exports, ["Stream"], () => require("./stream"));
utilities.lazyLoad(exports, ["StreamConsumer"], () => require("./streamConsumer"));
utilities.lazyLoad(exports, ["VideoStream"], () => require("./videoStream"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:kinesis/analyticsApplication:AnalyticsApplication":
                return new exports.AnalyticsApplication(name, undefined, { urn });
            case "aws:kinesis/firehoseDeliveryStream:FirehoseDeliveryStream":
                return new exports.FirehoseDeliveryStream(name, undefined, { urn });
            case "aws:kinesis/stream:Stream":
                return new exports.Stream(name, undefined, { urn });
            case "aws:kinesis/streamConsumer:StreamConsumer":
                return new exports.StreamConsumer(name, undefined, { urn });
            case "aws:kinesis/videoStream:VideoStream":
                return new exports.VideoStream(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "kinesis/analyticsApplication", _module);
pulumi.runtime.registerResourceModule("aws", "kinesis/firehoseDeliveryStream", _module);
pulumi.runtime.registerResourceModule("aws", "kinesis/stream", _module);
pulumi.runtime.registerResourceModule("aws", "kinesis/streamConsumer", _module);
pulumi.runtime.registerResourceModule("aws", "kinesis/videoStream", _module);
